File: app/src/main/java/ru/evgeniy/nytimes/news/StoryMappers.java
package ru.evgeniy.nytimes.news;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import ru.evgeniy.nytimes.data.db.NewsEntity;
import ru.evgeniy.nytimes.data.network.dto.MultimediaDTO;
import ru.evgeniy.nytimes.data.network.dto.NewsDTO;

public class StoryMappers {

    private StoryMappers() {
        throw new AssertionError("Must be no instance");
    }

    static List<NewsEntity> map(@NonNull List<NewsDTO> dtos) {

        List<NewsEntity> items = new ArrayList<>();


        for (int i = 0; i < dtos.size(); i++) {
            NewsEntity item = toDatabaseType(dtos.get(i));
            items.add(item);
        }
        return Collections.unmodifiableList(items);
    }

    @NonNull
    private static NewsEntity toDatabaseType(@NonNull NewsDTO dto) {
         String title = dto.getTitle();

         List<MultimediaDTO> multimedia = dto.getMultimedia();
         String imageUrl = mapImage(multimedia);
         String category = dto.getSubsection();
         String publishDate = dto.getPublished_date();
         String preview = dto.getAbstractX();
         String url = dto.getUrl();

        return new NewsEntity(title,
                imageUrl,
                category,
                publishDate,
                preview,
                preview,
                url);
    }

    @Nullable
    private static String mapImage(@Nullable List<MultimediaDTO> multimedias) {

        if (multimedias != null && !multimedias.isEmpty()) {
            return multimedias.get(multimedias.size()-1).getUrl();
        }
        return null;

    }
}

File: app/src/main/java/ru/evgeniy/nytimes/news/StoryMappers.kt
package ru.evgeniy.nytimes.news

import java.util.ArrayList
import java.util.Collections

import ru.evgeniy.nytimes.data.db.NewsEntity
import ru.evgeniy.nytimes.data.network.dto.MultimediaDTO
import ru.evgeniy.nytimes.data.network.dto.NewsDTO

class StoryMappers private constructor() {

    init {
        throw AssertionError("Must be no instance")
    }

    companion object {

        fun map(dto: List<NewsDTO>): List<NewsEntity> {

            val items = ArrayList<NewsEntity>()


            for (i in dto.indices) {
                val item = toDatabaseType(dto[i])
                items.add(item)
            }
            return Collections.unmodifiableList(items)
        }

        private fun toDatabaseType(dto: NewsDTO): NewsEntity {
            val title = dto.title

            val multimedia = dto.multimedia
            val imageUrl = mapImage(multimedia)
            val category = dto.subsection
            val publishDate = dto.published_date
            val preview = dto.abstractX
            val url = dto.url

            return NewsEntity(title,
                    imageUrl,
                    category,
                    publishDate,
                    preview,
                    preview,
                    url)
        }

        private fun mapImage(multimedia: List<MultimediaDTO>?): String? {

            return if (multimedia != null && !multimedia.isEmpty()) {
                multimedia[multimedia.size - 1].url
            } else null

        }
    }
}
